src/proficiency.cpp

```diff
@@ -332,6 +387,24 @@ std::vector<proficiency_id> proficiency_set::learning_profs() const
     return ret;
 }
 
+float proficiency_set::get_proficiency_bonus( std::string category,
+        proficiency_bonus_type prof_bonus ) const
+{
+    int stat_bonus = 0;
+    for( const proficiency_id &knows : known ) {
+        proficiency prof = knows.obj();
+
+        std::map<std::string, std::vector<proficiency_bonus>> bonuses = prof.get_bonuses();
+
+        for( size_t index = 0; index < bonuses[category].size(); index++ ) {
+            if( bonuses[category][index].type == prof_bonus ) {
+                stat_bonus += bonuses[category][index].value;
+            }
+        }
```

Comment:
And this simplifies to:
I changed stat_bonus to a float, because this function returns a float, and bonus.value is a float.
Also, `proficiency prof = knows.obj();` unecessarily copies the proficiency (so does the category argument, hence the change to the const reference)
```suggestion
float proficiency_set::get_proficiency_bonus( const std::string &category,
        proficiency_bonus_type prof_bonus ) const
{
    float stat_bonus = 0;
    for( const proficiency_id &knows : known ) {
        for( const proficiency_bonus &bonus : knows->get_bonuses( category ) ) {
            if( bonus.type == prof_bonus ) {
                stat_bonus += bonus.value;
            }
        }
    }
```